src/sql/workbench/contrib/connection/browser/connectionStatus.ts

```diff
@@ -58,17 +62,52 @@ export class ConnectionStatusbarItem extends Disposable implements IWorkbenchCon
 
 	// Update the connection status shown in the bar
 	private _updateStatus(): void {
-		let activeConnection = TaskUtilities.getCurrentGlobalConnection(this._objectExplorerService, this._connectionManagementService, this._editorService, this._logService);
+		let activeConnection: IConnectionProfile = TaskUtilities.getCurrentGlobalConnection(this._objectExplorerService, this._connectionManagementService, this._editorService, this._logService);
+		let connectionId: string = undefined;
+		let editorNewInfo: ConnectionManagementInfo = this.getCurrentActiveEditorInfo();
```

Comment:
is it possible for this to throw exception? if so, we should handle exception here.

Reply:
Nah, it returns the info or undefined.